comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -58,9 +64,16 @@ def show_key(le_key: QtWidgets.QLineEdit) -> None:
     # only file settings are saved
     if talker_key.file:
         # record the current row so we know where to add the button
-        btn_show_key = layout.rowCount()
+        btn_test_row = layout.rowCount()
         le_key = generate_textbox(talker_key, layout)
+
+        # Generate show/hide icon for key/password within the text box
         le_key.setEchoMode(QtWidgets.QLineEdit.EchoMode.PasswordEchoOnEdit)
+
+        le_key_toggle = le_key.addAction(visibleIcon, QtWidgets.QLineEdit.TrailingPosition)
+        le_key_toggle.setToolTip("Show/Hide")
+        le_key_toggle.triggered.connect(partial(show_key, le_key=le_key, le_action=le_key_toggle))
+
```

Comment:
> I think the best solution will then be this way: https://kushaldas.in/posts/creating-password-input-widget-in-pyqt.html

There should be a new `generate_password_textbox` that uses a class similar to the one created in the above link instead of patching this in after the fact.

Reply:
I copied it wholesale in https://github.com/comictagger/comictagger/pull/465/commits/057725c5da0e7c9032cf5985d8bcaab129431182